internal/cloudsql/instance.go

```diff
@@ -166,32 +186,39 @@ func (i *Instance) scheduleRefresh(d time.Duration) *refreshResult {
 	res.ready = make(chan struct{})
 	res.timer = time.AfterFunc(d, func() {
 		ctx, cancel := context.WithTimeout(context.Background(), i.refreshTimeout)
-		res.md, res.tlsCfg, res.err = performRefresh(ctx, i.client, i.clientLimiter, i.connName, i.key)
+		res.md, res.tlsCfg, res.expiry, res.err = performRefresh(ctx, i.client, i.clientLimiter, i.connName, i.key)
 		cancel()
 
 		close(res.ready)
 		// Once the refresh is complete, update "current" with working result and schedule a new refresh
 		i.resultGuard.Lock()
 		defer i.resultGuard.Unlock()
-		// TODO: only replace cur result if it's not valid
-		i.cur = res
+		// if failed, scheduled the next refresh immediately
 		if res.err != nil {
-			// TODO: add a backoff on retries
-			// if failed, scheduled the next refresh immediately
 			i.next = i.scheduleRefresh(0)
+			// keep using current info unless it's no longer valid
+			// TODO: consider how to avoid supressing errors here
```

Comment:
Essentially every time this fails and cur is still valid, the error isn't exposed to the user. This is potentially problematic because the first error might be different than the tenth error, and so if the 10th error ends up being something like "sqladmin api quota" then the root problem was likely an earlier error. 

So the options are:
1. Expose this via logging -- but since this is a library and not a process I'd prefer not injecting logs to my user's application. (We may have to go this route anyway, since it's probably a better experience in the proxy if the errors are surfaced immediately. )

2. Figure out a way to chain errors so that if they eventually bubble up together, but a successful refresh eliminates the chain. (This may still be subpar, as errors may still indicate that something is going wrong will never be surfaced) 

I couldn't really decide, so I left a TODO to think about it some more.